python/cudf/cudf/dataframe/buffer.py

```diff
@@ -126,14 +126,13 @@ def append(self, element):
         self.extend(np.asarray(element, dtype=self.dtype))
 
     def extend(self, array):
+        from cudf.dataframe import columnops
+
         needed = array.size
         self._sentry_capacity(needed)
-        import cudf.bindings.typecast as typecast
-        from cudf.dataframe import columnops
 
-        array = typecast.apply_cast(
-            columnops.as_column(array), dtype=self.dtype.type
-        ).data.mem
+        array = columnops.as_column(array).astype(self.dtype).data.mem
```

Comment:
Similar change could be made at two more places:

https://github.com/rapidsai/cudf/blob/branch-0.9/python/cudf/cudf/bindings/copying.pyx#L59
https://github.com/rapidsai/cudf/blob/branch-0.9/python/cudf/cudf/bindings/sort.pyx#L187

Reply:
Great catch! Thanks!